src/config.c

```diff
@@ -688,6 +719,63 @@ void configSetCommand(redisClient *c) {
 
         if (yn == -1) goto badfmt;
         server.aof_rewrite_incremental_fsync = yn;
+#ifdef USE_NVML
+    } else if (!strcasecmp(c->argv[2]->ptr, "aof-use-nvml")){
+        int enable = yesnotoi(o->ptr);
+
+        if (enable == -1) goto badfmt;
+
+        if (server.aof_use_nvml != enable) {
+            if (server.aof_state == REDIS_AOF_OFF) {
+                server.aof_use_nvml = enable;
+            } else if (server.aof_state == REDIS_AOF_ON) {
+                // - change aof-nvml-value
+                // - rewrite aof
+                server.aof_use_nvml = enable;
+                redisLog(REDIS_NOTICE,"Restarting AOF to switch NVML mode");
+                stopAppendOnly();
+                if (startAppendOnly() == REDIS_ERR) {
```

Comment:
Yes, either classic or NVML AOF is lost and it's rewritten according to the new setting. I believe it could stay this way, I was thinking to add appropriate warning in config file, that changing it in runtime will cause losing old AOF file and triggering rewrite. But I realized now there's another problem: user could trigger aof rewrite by hand when AOF is disbaled. Now if he used classic aof, changed setting aof_use_nvml followed by triggering rewrite redis will crash, because NVML pools are not initialized.